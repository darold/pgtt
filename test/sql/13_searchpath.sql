----
-- Regression test to Global Temporary Table implementation
--
-- Test on search_path forced by the extension
--
----
DROP EXTENSION pgtt;

SHOW search_path ;
create schema first_schema;
create schema second_schema;
SET search_path TO "$user",public,pg_catalog,fisrt_schema,second_schema;
SHOW search_path ;
CREATE EXTENSION pgtt;
SHOW search_path ;
SET search_path TO pg_catalog;
SHOW search_path ;
SET search_path TO public;
SHOW search_path ;

-- Test only pg_catalog in search_path.
DROP EXTENSION pgtt;
\c - -
SHOW search_path;
SET search_path TO pg_catalog;
SHOW search_path;
CREATE EXTENSION pgtt;
SHOW search_path;

-- Test the pg_catalog at end of search_path.
DROP EXTENSION pgtt;
\c - -
SET search_path TO "$user", public, pg_catalog;
SHOW search_path;
CREATE EXTENSION pgtt;
SHOW search_path;
