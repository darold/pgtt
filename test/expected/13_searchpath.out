----
-- Regression test to Global Temporary Table implementation
--
-- Test on search_path forced by the extension
--
----
DROP EXTENSION pgtt;
SHOW search_path ;
         search_path          
------------------------------
 "$user", public, pgtt_schema
(1 row)

create schema first_schema;
create schema second_schema;
SET search_path TO "$user",public,pg_catalog,fisrt_schema,second_schema;
SHOW search_path ;
                              search_path                              
-----------------------------------------------------------------------
 "$user", public, pg_catalog, fisrt_schema, second_schema, pgtt_schema
(1 row)

CREATE EXTENSION pgtt;
SHOW search_path ;
                              search_path                              
-----------------------------------------------------------------------
 "$user", public, pg_catalog, fisrt_schema, second_schema, pgtt_schema
(1 row)

SET search_path TO pg_catalog;
SHOW search_path ;
       search_path       
-------------------------
 pg_catalog, pgtt_schema
(1 row)

SET search_path TO public;
SHOW search_path ;
     search_path     
---------------------
 public, pgtt_schema
(1 row)

-- Test only pg_catalog in search_path.
DROP EXTENSION pgtt;
\c - -
SHOW search_path;
   search_path   
-----------------
 "$user", public
(1 row)

SET search_path TO pg_catalog;
SHOW search_path;
 search_path 
-------------
 pg_catalog
(1 row)

CREATE EXTENSION pgtt;
SHOW search_path;
       search_path        
--------------------------
  pgtt_schema, pg_catalog
(1 row)

-- Test the pg_catalog at end of search_path.
DROP EXTENSION pgtt;
\c - -
SET search_path TO "$user", public, pg_catalog;
SHOW search_path;
         search_path         
-----------------------------
 "$user", public, pg_catalog
(1 row)

CREATE EXTENSION pgtt;
SHOW search_path;
               search_path                
------------------------------------------
 "$user", public, pgtt_schema, pg_catalog
(1 row)

